portfolio/src/main/java/com/google/sps/data/HighestPriorityFirstScheduler.java

```diff
@@ -0,0 +1,114 @@
+        currentScheduleTime = availableTimeRange.start();
+      }
+
+      if (!currentScheduleTime
+          .plusSeconds(task.getDuration().getSeconds())
+          .isAfter(availableTimeRange.end())) {
+        ScheduledTask scheduledTask =
+            new ScheduledTask(task, currentScheduleTime, Optional.of(true));
+        scheduledTasks.add(scheduledTask);
+
+        // Delete the TimeRange that has been scheduled over so that different priority tasks
+        // won't be scheduled over the same time.
+        availableTimesGroup.deleteTimeRange(
+            TimeRange.fromStartEnd(
+                scheduledTask.getStartTime(),
+                scheduledTask
+                    .getStartTime()
+                    .plusSeconds(scheduledTask.getTask().getDuration().getSeconds())));
+        // After deleting a TimeRange, we must renew the list.
+        currentAvailableTimes = availableTimeRangesList(availableTimesGroup);
+
+        // Push the currentScheduleTime back by the amount that was scheduled.
+        currentScheduleTime = currentScheduleTime.plusSeconds(task.getDuration().getSeconds());
+        taskQueue.remove();
+
+        // If the next task's priority is different from the task that was just scheduled,
+        // then reset the availableTimesIndex, currentScheduleTime, and make the loop
+        // grab a TimeRange. We go back to the first availableTimeRange in order to schedule
+        // more tasks towards the beginning of the availableTimes.
+        if (!taskQueue.isEmpty()
+            && taskQueue.peek().getPriority().getPriority() != task.getPriority().getPriority()) {
+          availableTimesIndex = 0;
+          currentScheduleTime = workHoursStartTime;
+        }
+        // If the task's priority is not different, then we can simply continue running the
+        // the loop without retrieving the next availableTimeRange since there might be
+        // some time left over towards the end of the current one.
+      } else if (availableTimesIndex == currentAvailableTimes.size() - 1) {
+        // If we've reached the end of the availableTimeRange group and we can't schedule a task,
+        // then we can remove all the remaining tasks of equal priority since they will all be
+        // longer in duration therefore, they will not be able to be scheduled either.
+        if (!taskQueue.isEmpty()) {
+          availableTimesIndex = 0;
+          while (!taskQueue.isEmpty()
+              && taskQueue.peek().getPriority().getPriority() == task.getPriority().getPriority()) {
+            taskQueue.remove();
+          }
+        }
+      } else {
+        // Get the next TimeRange if the Task cannot be scheduled but we aren't at the final
+        // TimeRange.
+        availableTimesIndex++;
+      }
+    }
+
+    Collections.sort(scheduledTasks, sortByScheduledStartTimeAscending);
+    return scheduledTasks;
+  }
+
+  private List<TimeRange> availableTimeRangesList(TimeRangeGroup updatedTimeRangeGroup) {
```

Comment:
I would rename the method into `getAvailableTimeRangesList`.

Reply:
Done.